external-crates/move/crates/move-compiler/src/linters/mod.rs

```diff
@@ -30,30 +34,100 @@ pub enum LinterDiagnosticCategory {
     Sui = 99,
 }
 
+macro_rules! lints {
+    (
+        $(
+            ($enum_name:ident, $filter_name:expr, $code_msg:expr)
+        ),* $(,)?
+    ) => {
+        #[derive(PartialEq, Eq, Clone, Copy, Debug, Hash, PartialOrd, Ord)]
+        #[repr(u8)]
+        pub enum StyleCodes {
+            DontStartAtZeroPlaceholder,
+            $(
+                $enum_name,
+            )*
+        }
+
+        impl StyleCodes {
+            const fn code_and_message(&self) -> (u8, &'static str) {
+                let code = *self as u8;
+                debug_assert!(code > 0);
+                match self {
+                    Self::DontStartAtZeroPlaceholder =>
+                        panic!("ICE do not use placeholder error code"),
+                    $(Self::$enum_name => (code, $code_msg),)*
+                }
+            }
+
+            const fn code_and_filter_name(&self) -> (u8, &'static str) {
+                let code = *self as u8;
+                debug_assert!(code > 0);
+                match self {
+                    Self::DontStartAtZeroPlaceholder =>
+                        panic!("ICE do not use placeholder error code"),
+                    $(Self::$enum_name => (code, $filter_name),)*
+                }
+            }
+
+            const fn diag_info(&self) -> DiagnosticInfo {
+                let (code, msg) = self.code_and_message();
+                custom(
+                    LINT_WARNING_PREFIX,
+                    Severity::Warning,
+                    LinterDiagnosticCategory::Style as u8,
+                    code,
+                    msg,
+                )
+            }
+        }
+
+        const STYLE_WARNING_FILTERS: &[(u8, &str)] = &[
+            $(
+                StyleCodes::$enum_name.code_and_filter_name(),
+            )*
+        ];
+    }
+}
+
+// Example usage:
+lints!(
+    (
+        ConstantNaming,
+        "constant_naming",
+        "constant should follow naming convention"
+    ),
+    (UnneededReturn, "unneeded_return", "unneeded return"),
+);
```

Comment:
The goal was to automate these, but it will lead to a bit of churn with existing lints.